types/errors/abci.go

```diff
@@ -41,27 +41,29 @@ func ABCIInfo(err error, debug bool) (codespace string, code uint32, log string)
 
 // ResponseCheckTx returns an ABCI ResponseCheckTx object with fields filled in
 // from the given error and gas values.
-func ResponseCheckTx(err error, gw, gu uint64, debug bool) abci.ResponseCheckTx {
+func ResponseCheckTx(err error, gw, gu uint64, anteEvents []abci.Event, debug bool) abci.ResponseCheckTx {
 	space, code, log := ABCIInfo(err, debug)
 	return abci.ResponseCheckTx{
 		Codespace: space,
 		Code:      code,
 		Log:       log,
 		GasWanted: int64(gw),
 		GasUsed:   int64(gu),
+		Events:    anteEvents,
 	}
 }
 
 // ResponseDeliverTx returns an ABCI ResponseDeliverTx object with fields filled in
 // from the given error and gas values.
-func ResponseDeliverTx(err error, gw, gu uint64, debug bool) abci.ResponseDeliverTx {
+func ResponseDeliverTx(err error, gw, gu uint64, anteEvents []abci.Event, debug bool) abci.ResponseDeliverTx {
```

Comment:
These 2 functions are API breaking, so we cannot merge them back into 0.44.

We can create 2 new functions `Response{Check,Deliver}TxWithEvents`

Reply:
```suggestion
func ResponseDeliverTx(err error, gw, gu uint64, events []abci.Event, debug bool) abci.ResponseDeliverTx {
```